ledger/txtail.go

```diff
@@ -92,6 +102,14 @@ func (t *txTail) newBlock(blk bookkeeping.Block, delta StateDelta) {
 		txleases: delta.txleases,
 		proto:    config.Consensus[blk.CurrentProtocol],
 	}
+
+	proto := config.Consensus[blk.CurrentProtocol]
+	if _, ok := t.lastValid[rnd+basics.Round(proto.MaxTxnLife)+1]; !ok {
+		t.lastValid[rnd+basics.Round(proto.MaxTxnLife)+1] = make(map[transactions.Txid]struct{})
+	}
```

Comment:
That won't work correctly if proto.MaxTxnLife goes up. i.e. 1000->1500
in that case, you'll need a for loop in here to add all the missing elements.
( very unlikely to happen, but plausible )